src/persist/src/operators/stream.rs

```diff
@@ -1061,6 +1062,66 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn conditional_seal_frontier_advance_only_on_success() -> Result<(), Error> {
```

Comment:
just to make sure I understand the bug --- if previously the PendingSealFuture for a seal to time t held onto a capability at time (t - 1) instead of t, then we wouldn't have the bug right?

Reply:
or no, thats not even right
 the only data that should be available is data for times that have successfully been sealed. so even though a seal for t is pending, it doesn't necessarily mean that t - 1, or any other time, is available

Reply:
That latter one is correct, yes. Releasing for `t-1` without knowing anything else is wrong.